src/lib/process/index.ts

```diff
@@ -67,86 +87,119 @@ export const listTransforming = async (
   dryRun: boolean = false,
   options: Polkadot.Options = defaultOptions,
-  verbose: boolean = false,
-): Promise<Process.Result> => {
+  verbose: boolean = false
+): Promise<Process.ProcessResponse> => {
   try {
+    const { name, version, program } = processValidation.parse(processRaw)
 
-    const program: Process.Program = validate(userProgram)
-    const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
+    const processId = utf8ToHex(name)
     const polkadot: Polkadot.Polkadot = await createNodeApi(options)
     const currentVersion: number = await getVersion(polkadot, processId)
     const expectedVersion: number = currentVersion + 1
 
-    if (version > expectedVersion || version < currentVersion)
-      throw new VersionError(version, expectedVersion, name)
+    if (version > expectedVersion || version < currentVersion) throw new VersionError(version, expectedVersion, name)
 
     if (version === currentVersion) {
       const process = await getProcess(polkadot, processId, version)
 
-      if (program.length !== process.program.length) 
-        throw new ProgramError('existing: programs are different lengths', process) 
+      if (program.length !== process.program.length)
+        throw new ProgramError('existing: programs are different lengths', process)
 
       if (!program.every((step, i) => textify(step) === textify(process.program[i])))
-        throw new ProgramError('existing: program steps did not match', process) 
+        throw new ProgramError('existing: program steps did not match', process)
 
       return {
+        type: 'ok',
         message: `Skipping: process ${name} is already created.`,
-        process,
+        result: handleVerbose(
+          {
+            name,
+            version,
+            program,
+            status: process.status,
+          },
+          verbose
+        ),
       }
     }
-    
 
     if (dryRun)
       return {
-        process: null,
+        type: 'ok',
         message: 'Dry run: transaction has not been created',
-        name,
-        version: expectedVersion,
-        program,
+        result: handleVerbose(
+          {
+            name,
+            version,
+            program,
+            status: 'Enabled (dry-run)',
+          },
+          verbose
+        ),
       }
 
     return {
+      type: 'ok',
       message: `Transaction for new process ${name} has been successfully submitted`,
-      process: await createProcessTransaction(polkadot, processId, program, options, verbose),
+      result: handleVerbose(await createProcessTransaction(polkadot, processId, program, options), verbose),
     }
   } catch (err) {
     // err is basically from errors.ts or any exception
     // process errors will comntain specific messages and/or process
     // Promise<Process.Result> is in try {} and any exception is in catch {}
-    return err
+    if (err instanceof ProgramError || err instanceof VersionError || err instanceof ZodError) {
```

Comment:
just thinking it it would make sense to wrap all of these errors into another extend or drop them into array just for easier read or make it more granular, is just something for the lovely person "Future" but I've observed that we repeat this pattern across

Reply:
I agree. I was hoping to use a type union for this but it didn't work properly with typeof. I think the only way to do this well would be a to have a base class that covered these then the instanceOf would work as expected. That would have meant rethrowing a wrapped version of `ZodError` so I decided to just leave it be and deal with this monster